hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/token/TokenAccessorImpl.java

```diff
@@ -0,0 +1,405 @@
+
+        return isSpenderAnOperator != null && isSpenderAnOperator;
+    }
+
+    @Override
+    public Address ownerOf(final Address nft, long serialNo) {
+        final var tokenId = entityIdFromEvmAddress(nft);
+        final var ownerNum = nftRepository.findOwner(tokenId, serialNo);
+        if (ownerNum.isEmpty()) {
+            return Address.ZERO;
+        }
+        final var ownerEntity = EntityIdEndec.decode(ownerNum.get(), ACCOUNT);
+        return Address.wrap(Bytes.wrap(toEvmAddress(ownerEntity)));
+    }
+
+    @Override
+    public Address canonicalAddress(final Address addressOrAlias) {
+        return addressOrAlias;
+    }
+
+    @Override
+    public String metadataOf(final Address nft, long serialNo) {
+        final var tokenId = entityIdFromEvmAddress(nft);
+        final var metadata = nftRepository.findMetadata(tokenId, serialNo);
+        return metadata.map(String::new).orElse("");
+    }
+
+    @Override
+    public byte[] ledgerId() {
+        return properties.getNetwork().getLedgerId();
+    }
+
+    private Optional<EvmTokenInfo> getTokenInfo(final Address token) {
+        final var tokenEntityOptional = tokenRepository.findById(new TokenId(fromEvmAddress(token.toArray())));
+        final var entityOptional = entityRepository.findById(entityIdFromEvmAddress(token));
+
+        if (tokenEntityOptional.isEmpty() || entityOptional.isEmpty()) {
+            return Optional.empty();
+        }
+
+        final var tokenEntity = tokenEntityOptional.get();
+        final var entity = entityOptional.get();
+        final var ledgerId = ledgerId();
+        final var expirationTimeInSec = entity.getExpirationTimestamp() == null ? 0L :
+                entity.getExpirationTimestamp() / 1000000000;
+
+        final EvmTokenInfo evmTokenInfo = new EvmTokenInfo(
+                ledgerId,
+                tokenEntity.getSupplyType().ordinal(),
+                entity.getDeleted(),
+                tokenEntity.getSymbol(),
+                tokenEntity.getName(),
+                entity.getMemo(),
+                toAddress(tokenEntity.getTreasuryAccountId()),
+                tokenEntity.getTotalSupply(),
+                tokenEntity.getMaxSupply(),
+                tokenEntity.getDecimals(),
+                expirationTimeInSec);
+        evmTokenInfo.setAutoRenewPeriod(entity.getAutoRenewPeriod() != null ? entity.getAutoRenewPeriod() : 0);
+        evmTokenInfo.setDefaultFreezeStatus(tokenEntity.getFreezeDefault());
+        evmTokenInfo.setCustomFees(getCustomFees(token));
+        setEvmKeys(entity, tokenEntity, evmTokenInfo);
+        final var isPaused = tokenEntity.getPauseStatus().ordinal() == 1;
+        evmTokenInfo.setIsPaused(isPaused);
+
+        entityRepository.findById(entity.getAutoRenewAccountId())
+                .ifPresent(a -> evmTokenInfo.setAutoRenewAccount(toAddress(
+                        new EntityId(a.getShard(), a.getRealm(), a.getNum(), EntityType.ACCOUNT))));
```

Comment:
We should only do this logic if autoRenewAccountId is not null. And we don't need to do a database lookup.

```suggestion
        if (entity.getAutoRenewAccountId() != null) {
            var autoRenewAddress = toAddress(EntityId.of(entity.getAutoRenewAccountId(), EntityType.ACCOUNT));
            evmTokenInfo.setAutoRenewAccount(autoRenewAddress);
        }
```